File: src/Abstract/debug_info.cpp
#include "LIEF/Abstract/Binary.hpp"
#include "LIEF/Abstract/DebugInfo.hpp"

#include "logging.hpp"
#include "messages.hpp"

namespace LIEF {
namespace details {
class DebugInfo {};
}

// ----------------------------------------------------------------------------
// Abstract/Binary.hpp
// ----------------------------------------------------------------------------
DebugInfo* Binary::debug_info() const {
  LIEF_ERR(DEBUG_FMT_NOT_SUPPORTED);
  return nullptr;
}

// ----------------------------------------------------------------------------
// DebugInfo/DebugInfo.hpp
// ----------------------------------------------------------------------------
DebugInfo::DebugInfo(std::unique_ptr<details::DebugInfo>) :
    impl_(nullptr)
{}

DebugInfo::~DebugInfo() = default;

}

File: api/rust/crates/lief-ffi/src/generic/debug_info.rs
#[cxx::bridge]
pub mod ffi {
    unsafe extern "C++" {
        include!("LIEF/rust/Abstract/DebugInfo.hpp");

        type AbstracDebugInfo;

        fn find_function_address(
            self: &AbstracDebugInfo,
            name: &CxxString,
            is_set: Pin<&mut u32>,
        ) -> u64;
    }

    impl UniquePtr<AbstracDebugInfo> {}
}
